### PR TITLE
CSS-5987 integration testing for policy relation

### PR DESCRIPTION
**Pull request by AmberCharitos**
Add integration tests for the policy relation. The test:
- Deploys Ranger and PostgreSQL charms
- Relates Trino and Ranger
- Creates policies in Ranger
- Verifies these policies are applied to users accessing Trino

### CODE REVIEW

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -9,21 +9,52 @@
...
+POSTGRES_NAME = "postgresql-k8s"
 NGINX_NAME = "nginx-ingress-integrator"
 CONN_CONFIG = """connector.name=postgresql
 connection-url=jdbc:postgresql://example.host.com:5432/test
 connection-user=trino
 connection-password=trino
 """
+RANGER_NAME = "ranger-k8s"
+GROUP_MANAGEMENT = """\
+    trino-service:
```
Comment (mertalpt) [inline tests/integration/helpers.py:38]: You can either use `text.dedent` or write the string with one less level of indentation to make this prettier in logs and prints.

**Review by mertalpt — APPROVED** [review]

File: tests/integration/test_policy.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+# See LICENSE file for licensing details.
+
+"""Trino policy integration tests."""
+
+import logging
+import time
+
+import pytest
+import pytest_asyncio
+from conftest import deploy  # noqa: F401, pylint: disable=W0611
```
Comment (kelkawi-a) [inline tests/integration/test_policy.py:12]: Are you deploying the application differently to how it is deployed in `conftest.py`? If so, perhaps this import is not necessary. If you are, then the `deploy_ranger` fixture below is not necessary.

File: tests/integration/helpers.py [hunk 3]
```diff
@@ -9,21 +9,52 @@
...
+          - name: commercial-systems
+            description: commercial systems team
+"""
+TRINO_USER = "trino"
+TRINO_SERVICE = "trino-service"
+USER_WITH_ACCESS = "user1"
+USER_WITHOUT_ACCESS = "user2"
+GROUP_WITH_ACCESS = "commercial-systems"
+POLICY_NAME = "tpch - catalog, schema, table, column"
+TRINO_POLICY = "trino-k8s-policy"
```
Comment (kelkawi-a) [inline tests/integration/helpers.py:57]: `TRINO_POLICY_APP_NAME`?

**Review by kelkawi-a — APPROVED** [review]
LGTM

File: tests/integration/test_policy.py (hunk 2 above)
Comment (AmberCharitos) [reply tests/integration/test_policy.py]: Yes I am differently, have removed.

File: tests/integration/helpers.py (hunk 3 above)
Comment (AmberCharitos) [reply tests/integration/helpers.py]: Going with `TRINO_POLICY_NAME` as we have `POSTGRES_NAME` etc.

File: tests/integration/helpers.py (hunk 1 above)
Comment (AmberCharitos) [reply tests/integration/helpers.py:38]: Thank you - I will use `text.dedent` in any loggings. Though I don't believe I have any as of now.

**Pull request merged**